### ISSUE TITLE
Add missing SuperDirt parameters

### ISSUE THREAD

**Issue by yaxu** [tidalcycles/Tidal#623]
`fadeTime` is missing https://github.com/tidalcycles/Tidal/issues/618

While we're about it, lets check to see if others are missing

**Comment by telephon** [comment]
- [ ] cps = 1.0
- [ ] offset = 0.0
- [ ] begin = 0.0
- [ ] end = 1.0
- [ ] speed = 1.0
- [ ] accelerate
- [ ] pan = 0.5
- [ ] gain = 1.0
- [ ] overgain = 0.0
- [ ] cut = 0.0
- [ ] unit = \r
- [ ] n = \none // sample number or note
- [ ] octave = 5
- [ ] note
- [ ] midinote = #{ note ?  n + ( octave * 12) }
- [ ] freq = #{ midinote.value.midicps }
- [ ] delta = 1.0
- [ ] latency = 0.0
- [ ] lag = 0.0
- [ ] offset
- [ ] length = 1.0
- [ ] loop = 1.0
- [ ] dry = 0.0
- [ ] lock = 0 // if set to 1, syncs delay times with cps

- [ ] amp = 0.4
- [ ] fadeTime = 0.001
- [ ] channel
- [ ] delaytime
- [ ] delayfeedback
- [ ] delaySend
- [ ] delayAmp
- [ ] lock
- [ ] size
- [ ] room
- [ ] dry
- [ ] leslie
- [ ] lrate
- [ ] lsize
- [ ] rmsReplyRate
- [ ] rmsPeakLag

**Comment by yaxu** [comment]
* `delta` isn't there but is calculated and sent automatically, so I ticked
* `latency` isn't there - what does it do? There is already a `nudge` that I think isn't sent but tidal adds to the timestamp. Would `latency` just do the same but on the superdirt side? Then maybe we don't need both
* `offset` again seems a bit like a synonym for `latency` and `nudge`. It is in `params` anyway but I'm not sure what it does
* `lag` is there but is an alias for `lagogo`. I have no idea what a lagogo is, maybe left over from tidal-midi. I can remove that alias anyway. But again, what does `lag` do?
* The `delay` family - would be convenient to have a group alias with a param for each
* Same for the `leslie` family

The unticked ones need adding. I don't know what most of them do though

**Comment by telephon** [comment]
You can delay by either `lag` (absolute, in seconds) or `offset` (relative to cps). I don't know off my head what `nudge` results in – there is none on the dirt side.

It would be good to have name families, yes. Like `# leslie_rate` instead of `lrate`? There could be still shortcuts on the tidal side, but it would be good to make the superdirt side more verbose and less crowded.

The convention could be `<synthname><delimiter><parametername>`. Maybe as delimiter using `_` 

This will be a breaking change to those who use superdirt directly, but better now than ending up in a mess.

**Comment by yaxu** [comment]
I noticed we don't have a standard for case, e.g. `delaySend` vs `delayfeedback`. Perhaps we could make it case insensitive?

**Comment by telephon** [comment]
Doing this in supercollider costs electricity for all the rest, could it be done in tidal?

**Comment by yaxu** [comment]
Yes potentially, but then there are people using superdirt directly. Maybe best to standardise in superdirt, deprecating the old versions for a while.
This relates to your thought about `leslie_rate` vs `lrate`.. Which is more usual for supercollider, `leslierate`, `leslie_rate` or `leslieRate`? 
I guess `overgain` is in a different category, because `over` isn't an effect that `gain` is a parameter of, so could stay lowercase (although `gainOver` might make sense then, but doesn't feel right..).

**Comment by telephon** [comment]
> Maybe best to standardise in superdirt, deprecating the old versions for a while.

yes, that is good.



> This relates to your thought about `leslie_rate` vs `lrate`.. Which is more usual for supercollider, `leslierate`, `leslie_rate` or `leslieRate`?

Camelcase, `leslieRate` would be most idiomatic, both in SuperCollider and HasKell, right?

**Comment by telephon** [comment]
Only if we'd decided to add the synth name before it for keeping the namespace in order, we could use underscore. Like `lfo_maxFreq` or similar …
